src/python/pants/option/global_options.py

```diff
@@ -286,8 +286,10 @@ def register_bootstrap_options(cls, register):
                   'The `--pants-distdir` and `--pants-workdir` locations are inherently ignored.')
     register('--glob-expansion-failure', advanced=True,
              default=GlobMatchErrorBehavior.warn, type=GlobMatchErrorBehavior,
-             help="Raise an exception if any targets declaring source files "
-                  "fail to match any glob provided in the 'sources' argument.")
+             help="What to do when any filesystem specs cannot be found or any files in a target's "
+                  "`sources` argument may not be found. This happens when the files specified do "
+                  "not exist on your machine or when they are ignored by the `--pants-ignore` "
+                  "option.")
```

Comment:
Should the second message be included in the error message? When there are multiple warnings, then I think it should only be rendered once to avoid verbosity.